Resources/doc/adding_invitation_registration.rst

```diff
@@ -0,0 +1,318 @@
+        {
+            return 'acme_invitation_type';
+        }
+    }
+
+Create the custom data transformer:
+
+.. code-block:: php
+
+    <?php
+
+    namespace Acme\UserBundle\Form\DataTransformer;
+
+    use Acme\UserBundle\Entity\Invitation;
+    use Doctrine\ORM\EntityManager;
+    use Symfony\Component\Form\DataTransformerInterface;
+    use Symfony\Component\Form\Exception\UnexpectedTypeException;
+
+    /**
+     * Transforms an Invitation to an invitation code.
+     */
+    class InvitationToCodeTransformer implements DataTransformerInterface
+    {
+        protected $entityManager;
+
+        public function __construct(EntityManager $entityManager)
+        {
+            $this->entityManager = $entityManager;
+        }
+
+        public function transform($value)
+        {
+            if (null === $value) {
+                return null;
+            }
+
+            if (!$value instanceof Invitation) {
+                throw new UnexpectedTypeException($value, 'Acme\UserBundle\Entity\Invitation');
+            }
+
+            return $value->getCode();
+        }
+
+        public function reverseTransform($value)
+        {
+            if (null === $value || '' === $value) {
+                return null;
+            }
+
+            if (!is_string($value)) {
+                throw new UnexpectedTypeException($value, 'string');
+            }
+
+            return $this->entityManager
+                ->getRepository('Acme\UserBundle\Entity\Invitation')
+                ->findOneBy(array(
+                    'code' => $value,
+                    'user' => null,
+                ));
+        }
+    }
+
+Register your custom form type in the container:
+
+.. code-block:: xml
+
+    <!-- src/Acme/UserBundle/Resources/config/services.xml -->
+
+    <?xml version="1.0" ?>
+
+    <container xmlns="http://symfony.com/schema/dic/services"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+        <services>
+
+            <service id="acme.registration.form.type" class="Acme\UserBundle\Form\Type\RegistrationFormType">
+                <argument>%fos_user.model.user.class%</argument>
+                <tag name="form.type" alias="acme_user_registration" />
+            </service>
+
+            <service id="acme.invitation.form.type" class="Acme\UserBundle\Form\Type\InvitationFormType">
+                <argument type="service" id="acme.invitation.form.data_transformer"/>
+                <tag name="form.type" alias="acme_invitation_type" />
+            </service>
+
+            <service id="acme.invitation.form.data_transformer" class="Acme\UserBundle\Form\DataTransformer\InvitationToCodeTransformer">
+                <argument type="service" id="doctrine.orm.entity_manager"/>
+            </service>
+
+
+        </services>
+    </container>
+
+Or if you prefer the Yaml version:
```

Comment:
Can you use a configuration-block instead to provide the XML vs Yaml documentation ? This will be better once we have it on symfony.com, and I can render the doc on ReadTheDocs in the meantime to use the extension as well